src/util/makeCompareByProp.ts

```diff
@@ -1,16 +1,18 @@
 import { lower } from './lower'
 
+import { ComparatorFunction } from '../utilTypes'
+
 /** Returns true if the first object's key is greater than the second object's key. */
-export const isGreater = (a, b, key) => lower(a[key]) > lower(b[key])
+export const isGreater = (a:Object, b:Object, key: keyof typeof a & keyof typeof b )  => lower(a[key]) > lower(b[key])
```

Comment:
Can you explain the expression `keyof typeof a & keyof typeof b`?

Reply:
The intention behind adding this was to ensure that the key passed to this function belongs to both the objects, so it doesn't end up doing `lower(undefined) >lower(undefined)`
It didn't work as expected but it's fixed in [this](https://github.com/cybersemics/em/pull/656/commits/fb102a1254eee7ce8d99aa6157aa3606b16ce29b) commit